pkg/client/query/cache/memory.go

```diff
@@ -10,295 +10,369 @@ import (
 )
 
 var (
-	_ client.QueryCache[any]           = (*InMemoryCache[any])(nil)
-	_ client.HistoricalQueryCache[any] = (*InMemoryCache[any])(nil)
+	_ client.QueryCache[any]           = (*inMemoryCache[any])(nil)
+	_ client.HistoricalQueryCache[any] = (*inMemoryCache[any])(nil)
 
 	DefaultQueryCacheConfig = queryCacheConfig{
-		EvictionPolicy: FirstInFirstOut,
+		evictionPolicy: FirstInFirstOut,
 		// TODO_MAINNET(@bryanchriswhite): Consider how we can "guarantee" good
-		// alignment between the TTL and the block production rate.
-		TTL: time.Minute,
+		// alignment between the TTL and the block production rate,
+		// by accessing onchain block times directly.
+		ttl: time.Minute,
 	}
 )
 
-// InMemoryCache provides a concurrency-safe in-memory cache implementation with
+// inMemoryCache provides a concurrency-safe in-memory cache implementation with
 // optional historical value support.
-type InMemoryCache[T any] struct {
-	config       queryCacheConfig
-	latestHeight atomic.Int64
-
-	itemsMu sync.RWMutex
-	// items type depends on historical mode:
-	// | historical mode |  type                          |
-	// | --------------- | ------------------------------ |
-	// | false           | map[string]cacheItem[T]        |
-	// | true            | map[string]cacheItemHistory[T] |
-	items map[string]any
+type inMemoryCache[T any] struct {
+	config        queryCacheConfig
+	latestVersion atomic.Int64
```

Comment:
Per my other comment, can you #PUC why we need this and how its used?

Reply:
:saluting_face: :sweat_smile: